src/components/assignment/js/base-collection.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+var BaseCollection = function (models, options) {
+    this._models = [];
+    this.options = options;
+    this.initialize.apply(this, arguments);
+};
+
+_.extend(BaseCollection.prototype, {
+    idAttribute: 'id',
+
+    initialize: function (models, options) {
+    },
+
+    create: function (attr, options) {
+        var m = new this.model(attr, options);
+        this.set(m);
+        return m;
+    },
+
+    reset: function (models, options) {
+        this._models.length = 0;
+        this.set(models);
+    },
+
+    remove: function (model) {
+        _.remove(this._models, function (m) {
+            return m === model;
+        });
+
+        delete model.collection;
+
+        return model;
+    },
+
+    set: function (models) {
+        if (!models) {
+            return;
+        }
+
+        models = [].concat(models);
+
+        if (!models.length) {
+            return;
+        }
+
+        _.each(models, function (m) {
+            if (!(m instanceof this.model)) {
+                m = new this.model(m);
+            }
+
+            m.collection = this;
+
+            this._models.push(m);
+        }, this);
+
+        this.sort.call(this);
+
+        return models;
+    },
+
+    sortFn: function (a, b) {
+        return b._data[this.idAttribute] - a._data[this.idAttribute];
+    },
+
+    sort: function () {
+        this._models = this._models.sort(this.sortFn.bind(this));
+
+        return this._models;
+    },
+
+    getById: function (id) {
+        return _.find(this._models, function (m) {
+            return m.get(this.idAttribute) === id;
+        }, this);
+    },
+
+    each: function (cb, ctx) {
+        return _.each(this._models, cb, ctx || this);
+    },
+
+    all: function (cb, ctx) {
+        return _.all(this._models, cb, ctx || this);
+    },
+
+    toJSON: function () {
+        return _.invoke(this._models, 'toJSON');
+    },
+
+    find: function (fn) {
+        return _.find(this._models, fn);
+    },
+
+    filter: function (fn) {
+        return _.filter(this._models, fn);
+    },
+
+    length: function () {
```

Comment:
I think either `length` as a property or `size` as a function to follow backbone/jquery standards